tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersValidatorHelper.cs

```diff
@@ -0,0 +1,141 @@
+                var hasWritePermission = await ValidatePermissionsAsync(username);
+
+                var isValidCodeowner = hasRequiredOrgs && hasWritePermission;
+                return new CodeownersValidationResult
+                {
+                    Username = username,
+                    Status = "Success",
+                    Message = isValidCodeowner
+                        ? "Valid code owner" 
+                        : "Not a valid code owner",
+                    Organizations = organizations,
+                    HasWritePermission = hasWritePermission,
+                    IsValidCodeOwner = isValidCodeowner,
+                };
+            }
+            catch (NotFoundException)
+            {
+                return new CodeownersValidationResult
+                {
+                    Username = username,
+                    Status = "Error",
+                    Message = $"GitHub user not found '{username}' not found"
+                };
+            }
+            catch (RateLimitExceededException ex)
+            {
+                return new CodeownersValidationResult
+                {
+                    Username = username,
+                    Status = "Error",
+                    Message = $"Rate limit exceeded for user {username}. Reset at: {ex.Reset}"
+                };
+            }
+            catch (SecondaryRateLimitExceededException ex)
+            {
+                return new CodeownersValidationResult
+                {
+                    Username = username,
+                    Status = "Error",
+                    Message = $"Secondary rate limit exceeded for user {username}: {ex.Message}."
+                };
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error validating user {username}: {ex.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Validates the user's organization memberships (Microsoft and Azure only).
+        /// </summary>
+        private async Task<HashSet<string>> ValidateOrganizationsAsync(string username)
+        {
+            try
+            {
+                var organizations = await githubService.GetPublicOrgMembership(username);
+                var userOrgs = organizations.Select(org => org.Login).ToHashSet(StringComparer.OrdinalIgnoreCase);
+                return userOrgs;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error validating organizations for user: {Username}", username);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Validates the user's write permissions on azure-sdk-for-net repository.
+        /// </summary>
+        private async Task<bool> ValidatePermissionsAsync(string username)
+        {
+            try
+            {
+                var permission = await githubService.HasWritePermission("Azure", "azure-sdk-for-net", username);
```

Comment:
Add comment "Write access to Azure/azure-sdk-for-net is a sufficient proxy for knowing if the user has write permissions" or something like that.